### PR TITLE
[BUGFIX] initial-relationship-setup must handle null

### PR DESCRIPTION
**Pull request by runspired**

### CODE REVIEW

File: addon/-private/system/relationships/state/has-many.js [hunk 1]
```diff
@@ -180,13 +180,15 @@ export default class ManyRelationship extends Relationship {
...
 
   setInitialInternalModels(internalModels) {
-    let args = [0, this.canonicalState.length].concat(internalModels);
-    this.canonicalState.splice.apply(this.canonicalState, args);
-    internalModels.forEach(internalModel => {
-      this.canonicalMembers.add(internalModel);
-      this.members.add(internalModel);
-      this.setupInverseRelationship(internalModel);
-    });
+    if (internalModels && internalModels.length) {
```
Comment (igorT) [inline addon/-private/system/relationships/state/has-many.js]: Why would it be true but not have a length?

Comment (runspired) [inline addon/-private/system/relationships/state/has-many.js]: I was just being overly cautious

Comment (runspired) [inline addon/-private/system/relationships/state/has-many.js]: I have only ever hit undefined or an array with items in practice

Comment (igorT) [inline addon/-private/system/relationships/state/has-many.js]: If it's not undefined but has no length, l think we are in a bad state. We should assert against that in debug maybe, but we shouldn't guard, that would just hide a bug

Comment (runspired) [inline addon/-private/system/relationships/state/has-many.js]: Dropped the length check, I suspect asserting length here would be asserting at the wrong layer.

**Pull request merged**